include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -140,9 +232,13 @@ namespace internal
 
           if (needs_simplex_setup)
             {
-              quadrature_simplex = std::make_unique<Quadrature<dim>>(
-                Simplex::FE_P<dim, spacedim>(n_subdivisions)
-                  .get_unit_support_points());
+              if (dim == 2)
+                quadrature_simplex = std::make_unique<Quadrature<dim>>(
+                  generate_simplex_evaluation_points<dim>(n_subdivisions));
+              else
+                quadrature_simplex = std::make_unique<Quadrature<dim>>(
+                  Simplex::FE_P<dim, spacedim>(n_subdivisions)
+                    .get_unit_support_points());
```

Comment:
Is it true that you're basically just special casing the 2d case because it's reasonably simple to do and because creating the `FE_Q` is more expensive?